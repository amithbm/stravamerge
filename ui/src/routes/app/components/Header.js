import React from 'react';
import { Navbar, Nav, NavItem } from 'react-bootstrap';
import { Link } from 'react-router'
import { LinkContainer } from 'react-router-bootstrap'

export default ({isAuthenticated}) => {
    return <Navbar staticTop>
        <Navbar.Header>
            <Navbar.Brand>
                <Link to={isAuthenticated ? '/dashboard' : '/' }><span className="prefixlogocolor">ride</span><span className="postfixlogocolor">analysis</span></Link>
            </Navbar.Brand>
            <Navbar.Toggle />
        </Navbar.Header>
        <Navbar.Collapse>
            {isAuthenticated ?
                <Nav pullRight>
                    <LinkContainer to="/signOut"><NavItem eventKey={1}>Sign Out</NavItem></LinkContainer>
                </Nav> : undefined
            }
        </Navbar.Collapse>
    </Navbar>
}