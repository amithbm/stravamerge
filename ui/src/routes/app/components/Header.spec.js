import React from 'react'
import Header from './Header'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('components', () => {
    describe('<Header />', () => {
        it('renders with a sign out link when authenticated', () => {
            const component = shallow(<Header isAuthenticated={true} />)
            const tree = toJson(component)
            expect(tree).toMatchSnapshot()
        })
    })

    describe('<Header />', () => {
        it('renders with no sign out link when unauthenticated', () => {
            const component = shallow(<Header isAuthenticated={false} />)
            const tree = toJson(component)
            expect(tree).toMatchSnapshot()
        })
    })
})
