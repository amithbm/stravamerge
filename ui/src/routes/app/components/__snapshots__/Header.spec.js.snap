exports[`components <Header /> renders with a sign out link when authenticated 1`] = `
<Uncontrolled(Navbar)
  bsStyle="default"
  staticTop={true}>
  <NavbarHeader>
    <NavbarBrand>
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/dashboard">
        <span
          className="prefixlogocolor">
          ride
        </span>
        <span
          className="postfixlogocolor">
          analysis
        </span>
      </Link>
    </NavbarBrand>
    <NavbarToggle />
  </NavbarHeader>
  <NavbarCollapse>
    <Nav
      bsClass="nav"
      justified={false}
      pullLeft={false}
      pullRight={true}
      stacked={false}>
      <LinkContainer
        action="push"
        onlyActiveOnIndex={false}
        to="/signOut">
        <NavItem
          active={false}
          disabled={false}
          eventKey={1}>
          Sign Out
        </NavItem>
      </LinkContainer>
    </Nav>
  </NavbarCollapse>
</Uncontrolled(Navbar)>
`;

exports[`components <Header /> renders with no sign out link when unauthenticated 1`] = `
<Uncontrolled(Navbar)
  bsStyle="default"
  staticTop={true}>
  <NavbarHeader>
    <NavbarBrand>
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/">
        <span
          className="prefixlogocolor">
          ride
        </span>
        <span
          className="postfixlogocolor">
          analysis
        </span>
      </Link>
    </NavbarBrand>
    <NavbarToggle />
  </NavbarHeader>
  <NavbarCollapse />
</Uncontrolled(Navbar)>
`;
